build failed with -DENABLE_LLVM=0

build failed with -DENABLE_LLVM=0

/home/liucong/working/pocl/pocld/shared_cl_context.cc:1124:19: error: ‘CLANG’ was not declared in this scope
 1124 |     OpenCLCCmd << CLANG
      |                   ^~~~~
/home/liucong/working/pocl/pocld/shared_cl_context.cc:1141:15: error: ‘LLVM_SPIRV’ was not declared in this scope
 1141 |     SpvCmd << LLVM_SPIRV << " -r " << OrigSpvFileName.c_str() << " -o "
      |               ^~~~~~~~~~
/home/liucong/working/pocl/pocld/shared_cl_context.cc:1162:13: error: ‘LLVM_OPT’ was not declared in this scope
 1162 |   OptCmd << LLVM_OPT << " -load-pass-plugin=" << LibPoCLPath
      |             ^~~~~~~~
/home/liucong/working/pocl/pocld/shared_cl_context.cc:1173:13: error: ‘LLVM_SPIRV’ was not declared in this scope
 1173 |   SpvCmd << LLVM_SPIRV << " " << OffsettedBcFileName.c_str() << " -o "
      |             ^~~~~~~~~~

_Originally posted by @xvim in https://github.com/pocl/pocl/issues/1408#issuecomment-1952030329_

--- a/pocld/shared_cl_context.cc
+++ b/pocld/shared_cl_context.cc
@@ -1074,9 +1074,10 @@ int SharedCLContext::freeQueue(uint32_t queue_id) {
   buf += len;                                                                  \
   assert((size_t)(buf - buffer) <= buffer_size);
 
+#if defined(CLANG) && defined(LLVM_SPIRV)
 /**
- * Creates a SPIRV with all global memory addresses adjusted by adding the
- * SVMOffset.
+ * Creates a SPIRV with all global memory addresses adjusted by adding
+ * the SVMOffset.
  *
  * Used to adjust to an SVM region offset difference between the device and the
  * host. SVMOffset is asumed to wrap around the addition at unsigned 64b for
@@ -1186,6 +1187,7 @@ bool createSPIRVWithSVMOffset(const std::vector<unsigned char> *InputSPV,
   // std::filesystem::remove_all(TempDir);
   return true;
 }
+#endif
 
 int SharedCLContext::buildOrLinkProgram(
     uint32_t program_id, std::vector<uint32_t> &DeviceList, char *src,
@@ -1270,10 +1272,14 @@ int SharedCLContext::buildOrLinkProgram(
     std::vector<char> SVMOffsettedSPIRV;
 
 
+#if defined(CLANG) && defined(LLVM_SPIRV)
     // Adjust the SVM region offset to the kernel code.
     bool SuccessfulOffsetting = createSPIRVWithSVMOffset(
         is_spirv ? &(*InputBinaries.begin()).second : nullptr, src, src_size,
         SVMRegionOffset, SVMOffsettedSPIRV, options);
+#else
+    bool SuccessfulOffsetting = false;
+#endif
 
     assert(SuccessfulOffsetting && SVMOffsettedSPIRV.size() > 0);
 

